### DISCUSSION TITLE
Individual color and possibly size for each border of a window

### DISCUSSION THREAD

**Discussion by thingsiplay** [qtile/qtile#3504] [Ideas]
Is it possible to have different border colors for each side of the windows? I am not sure how to do this in Qtile itself, if possible. I imagine new set of attributes like `border_normal_bottom` to "overwrite" the bottom portion of `border_normal`. Similar to what is possible with CSS on the web.

The reason why I come up with this is, because I am currently trying out an external downloaded layout called "Plasma", that tries to replicate i3wm style of window management: https://github.com/numirias/qtile-plasma . And from my previous experience with i3wm (before landing on Qtile) I remember there was an indicator on the right or bottom side where the next window would appear. A key mapping allows for switching between horizontal and vertical mode and a visual indicator would be useful. Especially for people coming from i3wm.

If this is otherwise possible, please tell me. I could not find a solution, hence this post.

**Comment by m-col** [comment]
It is not possible to do this with Qtile, but it _is_ possible in theory. However I don't think we would merge such a feature as these kinds of things can get tricky. See here for a discussion on the mechanics of implementing this kind of arbitrary rendering: https://github.com/qtile/qtile/discussions/3322

**Comment by elParaguayo** [comment]
I'd been thinking about this. I think an alternative approach could be to use a widget to indicate the side where the next window would open. It's a far from perfect idea though.

**Comment by thingsiplay** [comment]
I can fully understand why you are unwilling to put new risky code for a probably niche need that is not important. The alternate approach as a widget on the bar was also my idea if the original idea is not possible. My idea is to implement this as a boolean value for every window and update the widget when a window gets focus or the keys for changing the mode is pressed.

I am fairly new to creating widgets and want to explore how to do this in the near future. For the moment I am not able to create such a widget.
